### PR TITLE
Exclude some more dlls from packaging

### LINKED ISSUE #7290: Fix Windows installer's extra Dll's from GHA
see https://discourse.cmake.org/t/cpack-adds-weird-dlls-to-the-installer/9667

### PR DESCRIPTION
**Pull request by timosachsenberg**
## Checklist
- [ ] Make sure that you are listed in the AUTHORS file
- [ ] Add relevant changes and new features to the CHANGELOG file
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] New and existing unit tests pass locally with my changes
- [ ] Updated or added python bindings for changed or new classes (Tick if no updates were necessary.)

### How can I get additional information on failed tests during CI
<details>
  <summary>Click to expand</summary>
If your PR is failing you can check out

- The details of the action statuses at the end of the PR or the "Checks" tab.
- http://cdash.openms.de/index.php?project=OpenMS and look for your PR. Use the "Show filters" capability on the top right to search for your PR number.
  If you click in the column that lists the failed tests you will get detailed error messages.

</details>

### Advanced commands (admins / reviewer only)
<details>
  <summary>Click to expand</summary>
  
- `/reformat` (experimental) applies the clang-format style changes as additional commit. Note: your branch must have a different name (e.g., yourrepo:feature/XYZ) than the receiving branch (e.g., OpenMS:develop). Otherwise, reformat fails to push.
- setting the label "NoJenkins" will skip tests for this PR on jenkins (saves resources e.g., on edits that do not affect tests)
- commenting with `rebuild jenkins` will retrigger Jenkins-based CI builds
  
</details>

---
:warning: Note: Once you opened a PR try to minimize the number of *pushes* to it as every push will trigger CI (automated builds and test) and is rather heavy on our infrastructure (e.g., if several pushes per day are performed).

### CODE REVIEW

File: cmake/package_general.cmake [hunk 1]
```diff
@@ -69,7 +69,7 @@ set(OPENMS_LOGOSMALL ${PROJECT_SOURCE_DIR}/cmake/MacOSX/${OPENMS_LOGOSMALL_NAME}
 # On Windows we need to tell CMake where to look for.
 # We also do not need API sets. So exclude them.
 if(WIN32)
-  set(EXCLUDE "api-ms" "ext-ms" "hvsi" "pdmutilities" "wpaxholder")
+  set(EXCLUDE "api-ms" "ext-ms" "hvsi" "pdmutilities" "wpaxholder" "dxgi" "UxTheme" "d3d11" "WINNMM")
```
Comment (timosachsenberg) [inline cmake/package_general.cmake]:
```suggestion
  set(EXCLUDE "api-ms" "ext-ms" "hvsi" "pdmutilities" "wpaxholder" "dxgi" "uxtheme" "d3d11" "winnmm" "wldp")
```

**Comment by jpfeuffer** [comment]
please use lowercase.
https://cmake.org/cmake/help/latest/command/file.html#get-runtime-dependencies:~:text=DLL%20dependency%20names%20are%20converted%20to%20lowercase%20for%20matching%20filters.

also maybe a folder-based exclude might make sense. Not sure where they all are located. Probably better if someone tests it.

**Comment by timosachsenberg** [comment]
@JeeH-K can you give this a try after it is merged? (create installer and try to execute)

**Comment by timosachsenberg** [comment]
I think all of them are in the bin folder. Let's test.

**Comment by jpfeuffer** [comment]
Yes, but after install ;) this excludes based on the location before installing. But yes, just test it.

**Comment by JeeH-K** [comment]
@timosachsenberg did it work with your branch?
I tried to merge the corresponding develop branch to mine (https://github.com/OpenMS/OpenMS/commits/FDdevelop/) and tried GHA, still failed.
Installation was successful, yet gave an error message while trying to open ToppView
Screenshot 2024-02-07 at 4 05 40 PM

**Comment by timosachsenberg** [comment]
It never failed on my machine, I think. @JeeH-K I guess you don't have a windows development environment set up (me neither). @cbielow if you have one set up can you check in which folder these dlls are before installation? See Julianus' comment

**Comment by timosachsenberg** [comment]
Works on my Computer. Windows 11.

File: cmake/package_general.cmake [hunk 2]
```diff
@@ -69,7 +69,7 @@ set(OPENMS_LOGOSMALL ${PROJECT_SOURCE_DIR}/cmake/MacOSX/${OPENMS_LOGOSMALL_NAME}
 # On Windows we need to tell CMake where to look for.
 # We also do not need API sets. So exclude them.
 if(WIN32)
-  set(EXCLUDE "api-ms" "ext-ms" "hvsi" "pdmutilities" "wpaxholder")
+  set(EXCLUDE "api-ms" "ext-ms" "hvsi" "pdmutilities" "wpaxholder" "dxgi" "uxtheme" "d3d11" "winnmm" "wldp")
```
Comment (jpfeuffer) [inline cmake/package_general.cmake:72]:
```suggestion
  set(EXCLUDE "api-ms" "ext-ms" "hvsi" "pdmutilities" "wpaxholder" "dxgi" "uxtheme" "d3d11" "winmm" "wldp")
```

**Review by jpfeuffer — CHANGES_REQUESTED** [review]

File: cmake/package_general.cmake (hunk 2 above)
Comment (cbielow) [reply cmake/package_general.cmake:72]: that should do it.

Comment (timosachsenberg) [reply cmake/package_general.cmake:72]: can add

**Comment by cbielow** [comment]
> It never failed on my machine, I think. @JeeH-K I guess you don't have a windows development environment set up (me neither). @cbielow if you have one set up can you check in which folder these dlls are before installation? See Julianus' comment

just for completeness, on my machine `winmm.dll` is in C:\Windows\System32
But your system should have it too. Its a very generic dll (Windows Multimedia API)

File: cmake/package_general.cmake (hunk 2 above)
Comment (timosachsenberg) [reply cmake/package_general.cmake:72]: Pushed. @JeeH-K

**Comment by JeeH-K** [comment]
Mine is Windows 10. It worked like a charm after the change from @jpfeuffer! Thanks everyone 👏

**Pull request merged**